Intersect.Client.Core/Interface/Game/Spells/SpellItem.cs

```diff
@@ -145,43 +129,66 @@ private void _iconImage_HoverEnter(Base? sender, EventArgs? arguments)
         _descriptionWindow = new SpellDescriptionWindow(spellSlots[SlotIndex].Id, _spellWindow.X, _spellWindow.Y);
     }
 
-    private void _iconImage_HoverLeave(Base sender, EventArgs arguments)
+    private void IconImage_HoverLeave(Base sender, EventArgs arguments)
     {
-        _mouseOver = false;
-        _mouseX = -1;
-        _mouseY = -1;
+        _descriptionWindow?.Dispose();
+        _descriptionWindow = null;
+    }
 
-        if (_descriptionWindow != null)
+    private void IconImage_Clicked(Base sender, MouseButtonState arguments)
+    {
+        if (arguments.MouseButton is MouseButton.Right)
         {
-            _descriptionWindow.Dispose();
-            _descriptionWindow = null;
+            if (ClientConfiguration.Instance.EnableContextMenus)
+            {
+                OpenContextMenu();
+            }
+            else
+            {
+                Globals.Me?.TryForgetSpell(SlotIndex);
+            }
         }
     }
 
-    private void _iconImage_Clicked(Base sender, MouseButtonState arguments)
+    private void IconImage_DoubleClicked(Base sender, MouseButtonState arguments)
+    {
+        Globals.Me?.TryUseSpell(SlotIndex);
+    }
+
+    #endregion
+
+    #region Drag and Drop
+
+    public override bool DragAndDrop_HandleDrop(Package package, int x, int y)
     {
-        switch (arguments.MouseButton)
+        var targetNode = Interface.FindComponentUnderCursor(NodeFilter.None);
+
+        // Find the first parent acceptable in that tree that can accept the package
+        while (targetNode != default)
         {
-            case MouseButton.Left:
-                _clickTime = Timing.Global.MillisecondsUtc + 500;
-                break;
+            switch (targetNode)
+            {
+                case SpellItem spellItem:
+                    Globals.Me?.SwapSpells(SlotIndex, spellItem.SlotIndex);
```

Comment:
Cache `Globals.Me` and return false _before_ `var targetNode = ...;` if it's null